### PR TITLE
Farrah Rios React Calculator App

### PR DESCRIPTION
**Pull request by rios-isabelle-farrah**

### CODE REVIEW

**Comment by rios-isabelle-farrah** [comment]
updated calculator application to meet requirements for full 18 points.

File: calcculator-app/src/App.css [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.App {
+  display:flex;
+  justify-content:center;
+  /* text-align: center; */
+  align-items: center;
+}
+
```
Comment (mckiernantim) [inline calcculator-app/src/App.css:7]: It's always a good idea to clean up any unused css rules like line 4

File: calcculator-app/src/components/Grid.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from "react"
+import "./styles/Grid.css"
+
+
```
Comment (mckiernantim) [inline calcculator-app/src/components/Grid.js:4]: Lets try to run our code through a code formatter to make it a little easier to read.

File: calcculator-app/src/components/Input.js [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+    firstDisplay: [],
+    mod: "",
+    secondDisplay: [],
+    answer: 0,
+    // equal:
+  };
+
+  handleInputChange = (event) => {
+    //let display = event.target.value.split("");
+
```
Comment (mckiernantim) [inline calcculator-app/src/components/Input.js]: Looks good Farrah, but lets remember to remove debugger statements and commented out code for deployment

File: calcculator-app/src/components/Grid.js (hunk 2 above)
Comment (mckiernantim) [reply calcculator-app/src/components/Grid.js:4]: We have a lot of whitespace here and could really use a code formatter to make it easier to read.

File: calcculator-app/src/components/Input.js [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+      this.setState((prevState) => ({
+        firstDisplay: parseInt([...prevState.input]), //parseInt removes zero
+      }));
+    }
+  };
+
+  handleGrid = (event) => {
+    let num = 56789345667
+    console.log(num.toLocaleString())
+    console.log(this.state);
```
Comment (mckiernantim) [inline calcculator-app/src/components/Input.js:43]: What are we trying to accomplish with our `num` variable on 41?  If this is a test of `toLocaleString()` we should remember to remove it before pushing.

**Review by mckiernantim — COMMENTED** [review]
1. I noticed that when updating user input with negative numbers we would return NaN under certain circumstances.  I'm going to attach a screen shot of what happened but I'm not sure where or how it came about.
Screen Shot 2021-06-08 at 3 31 23 PM

2. clean-up.  We have a good amount console.logs and debuggers and commented out code.  Remember to clean all this up before deploying as we don't want this in a production environment.  Remember when we deploy code the reason we want it clean is so other developers can read it and add to it or whatever else they need to do!
3. One thing I noticed is that the layout of your calculator numbers seems to go against the normal convention of 3 X 3 rows.  Its always a good idea to consider what your user is expecting when designing a front-end.  Will our users prefer your layout for this or will they like something more traditional?

File: calcculator-app/src/components/Input.js [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+        firstDisplay: [...[answer]].join(""),
+        previous: prevState,
+        mod: "",
+      }));
+    }
+    if (event.mod === "divide") {
+      let answer = Number(event.first) / Number(event.second);
+      this.setState((prevState) => ({
+        input: answer.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ","),
+        firstDisplay: [...[answer]].join(""),
```
Comment (mckiernantim) [inline calcculator-app/src/components/Input.js:219]: If we are going to repeat the use of the our regex on line 217 we should store it in a variable to keep our code DRYer.

File: calcculator-app/src/components/Mod.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    subtract: "-",
+    divide: "÷",
+    multiply: "*",
+    plusMinus: "+/-",
+ 
+  };
+
+  return (
+
+
```
Comment (mckiernantim) [inline calcculator-app/src/components/Mod.js:28]: This component looks great!  Let's look at removing whitespace for our production environment moving forward.

File: calcculator-app/src/components/Input.js [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+      input: [0],
+      firstDisplay: [],
+      mod: "",
+      secondDisplay: [],
+      answer: 0,
+    }));
+  };
+
+  handleEqual = (event) => {
+    console.log(event);
```
Comment (mckiernantim) [inline calcculator-app/src/components/Input.js:194]: This is some solid logic you've got here Farrah and you should be proud of your JS skills getting this far.  However, this has a lot of repetition inside of your `else` blocks.  I'd love to touch base at some point during flex or office hours to explore how we can get this code even DRYer by writing functions that we can call to handle our else blocks.

**Comment by rios-isabelle-farrah** [comment]
Thank you for the feedback, I didn't realize that this was the email that
all of my feedback was going to so I will make sure to check it more
diligently. I will remember to bring this up the next time I schedule
office hours because I do feel like there was a lot of tedious repetition
in my code but I couldnt figure out any other way to get the results
especially when it came implementing the commas.

**Pull request closed without merging**